client/src/app/pages/applications/analysis-wizard/analysis-wizard.tsx

```diff
@@ -163,10 +194,9 @@ export const AnalysisWizard: React.FC<IAnalysisWizard> = ({
   const methods = useForm<AnalysisWizardFormValues>({
     defaultValues: {
       artifact: null,
-      mode: "source-code-deps",
+      mode: determineMode(applications),
```

Comment:
```suggestion
      mode: determineMode(applications) ?? "source-code-deps",
```
That way, if a mode can't be figured out from the selected apps, the fallback/default mode is the original default.

Reply:
@sjd78 unfortunately @yael-spinner  hit a real bug -  the `AnalysisWizard` gets re-rendered with every change on the applications screen which means that the defaults are being set at start time when no apps are selected.
So `useEffect` is necessary here as a workaround. The good news is that using (our old friend) `key` property we can force re-creating the wizard when selected apps change. The following fix worked for me:
```
iff --git a/client/src/app/pages/applications/applications-table/applications-table.tsx b/client/src/app/pages/applications/applications-table/applications-table.tsx
index d102d69a..d4affbb4 100644
--- a/client/src/app/pages/applications/applications-table/applications-table.tsx
+++ b/client/src/app/pages/applications/applications-table/applications-table.tsx
@@ -1060,6 +1060,7 @@ export const ApplicationsTable: React.FC = () => {
 
         <TaskGroupProvider>
           <AnalysisWizard
+            key={selectedRows.map(({ name }) => name).join()}
             applications={selectedRows}
             isOpen={isAnalyzeModalOpen}
             onClose={() => {
(END)

```

Reply:
Ah right!  I forgot about the order state gets loaded up in modals like this.  Using the `key` prop does make sense.

For some reference: https://react.dev/learn/preserving-and-resetting-state#resetting-a-form-with-a-key

So use the `key` attribute and drop `useEffect()`?

Reply:
@sjd78  @rszwajko I changed the code according to your comments